main.tf

```diff
@@ -1,3 +1,280 @@
+  operator_target_namespace_msg      = "if input var ws_liberty_operator_target_namespace is null the value of ws_liberty_operator_namespace must be equal to ${local.default_liberty_operator_namespace}"
+  # tflint-ignore: terraform_unused_declarations
+  operator_target_namespace_chk = regex("^${local.operator_target_namespace_msg}$", (!local.operator_target_namespace_cnd ? local.operator_target_namespace_msg : ""))
+}
+
+data "ibm_container_cluster_config" "cluster_config" {
+  cluster_name_id = var.cluster_id
+  config_dir      = "${path.module}/kubeconfig"                                                             # See https://github.ibm.com/GoldenEye/issues/issues/552
+  endpoint_type   = var.cluster_config_endpoint_type != "default" ? var.cluster_config_endpoint_type : null # null represents default
+}
+
+# creating the namespace to deploy the helm releases to install the WebSphere Liberty Operator
+resource "kubernetes_namespace" "helm_release_operator_namespace" {
+  count = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  metadata {
+    name = var.operator_helm_release_namespace
+  }
+
+  timeouts {
+    delete = "30m"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      metadata[0].annotations,
+      metadata[0].labels
+    ]
+  }
+}
+
+locals {
+  ibm_operator_catalog_version = "v1.25-20240202.161709-9DAF3E648@sha256:92e28be4af60f68c656f52b2445aafcc052fcd0390479b868c5b0ba2d465a25a"
+  ibm_operator_catalog_path    = "icr.io/cpopen/ibm-operator-catalog"
+}
+
+# if add_ibm_operator_catalog is true going on with adding the IBM Operator Catalog source
+resource "helm_release" "ibm_operator_catalog" {
+  depends_on = [data.ibm_container_cluster_config.cluster_config, kubernetes_namespace.helm_release_operator_namespace[0]]
+  count      = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  name              = "ibm-operator-catalog-helm-release"
+  chart             = "${path.module}/chart/${local.ibm_operator_catalog_chart}"
+  namespace         = var.operator_helm_release_namespace
+  create_namespace  = false
+  timeout           = 300
+  dependency_update = true
+  force_update      = false
+  cleanup_on_fail   = false
+  wait              = true
+
+  disable_openapi_validation = false
+
+  set {
+    name  = "image.path"
+    type  = "string"
+    value = local.ibm_operator_catalog_path
+  }
+  set {
+    name  = "image.version"
+    type  = "string"
+    value = local.ibm_operator_catalog_version
+  }
+}
+
+# waiting for the catalog to be configured and correctly pulled
+resource "time_sleep" "wait_catalog" {
+  depends_on = [helm_release.ibm_operator_catalog[0]]
+  count      = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  create_duration = local.sleep_time_catalog_create
+}
+
+# if ws_liberty_operator_target_namespace != null the operator group must be created
+resource "helm_release" "websphere_liberty_operator_group" {
+  count      = var.ws_liberty_operator_target_namespace != null ? 1 : 0
```

Comment:
is that count right? If `ws_liberty_operator_target_namespace` is set to `null` it wont deploy this helm chart?

Reply:
yes, the operator group is needed only in the case of the operator target namespace != null, which means the SingleNamespace or OwnNamespace operator modes. While if the target namespace == null, the All Namespaces mode, it doesn't need any operator group

https://www.ibm.com/docs/en/was-liberty/core?topic=operator-installing-red-hat-openshift-cli#in-t-cli__install-op-cli__title__1